rs/recovery/src/steps.rs

```diff
@@ -509,34 +510,63 @@ impl Step for ValidateReplayStep {
 
 pub struct UploadAndRestartStep {
     pub logger: Logger,
-    pub node_ip: IpAddr,
+    pub upload_method: UploadMethod,
     pub work_dir: PathBuf,
     pub data_src: PathBuf,
     pub require_confirmation: bool,
     pub key_file: Option<PathBuf>,
     pub check_ic_replay_height: bool,
 }
 
+impl UploadAndRestartStep {
+    const CMD_STOP_REPLICA: &str = "sudo systemctl stop ic-replica;";
+    // Note that on older versions of IC-OS this service does not exist.
+    // So try this operation, but ignore possible failure if service
+    // does not exist on the affected version.
+    const CMD_RESTART_REPLICA: &str = "\
+        (sudo systemctl restart setup-permissions || true);\
+        sudo systemctl start ic-replica;\
+        sudo systemctl status ic-replica;";
+
+    /// Sets the right state permissions on `src`, by copying the permissions
+    /// of the target path, removing executable permission and giving read
+    /// permissions for the state to group and others.
+    fn cmd_set_permissions(target_path: &str, src: &str) -> String {
```

Comment:
Intuitively I would have expected the "src" to be the reference directory and "target" the directory we want to change permissions of, but it makes sense with the comment

Reply:
Reading it again, I agree it should be flipped. Done.